### PR TITLE
182: Declare Pre- and Post-Conditions on existing code

### LINKED ISSUE #182: Add pre and post conditions to functions and methods
This came out of our ID3 feedback. Dr. Osgood wants more explicit and verbose pre and post-conditions. Let's add them.

### PR DESCRIPTION
**Pull request by matthew-buglass**
Closes #182 

# What was the issue
- We had very few pre- and post-conditions
- Many of our doc strings needed beefing up
- Updated the project inspection profile to align with our ESLint line spacing
- Not Done yet. Just wanted to get some visibility

# What was done and why
- Added JSDoc style docstrings to our methods
- Added an inspection profile to warn us if our line limit is beyond what our ESLint prescribes
- Removed dead code

# How it was tested
- Launched the app and played around making sure no functionality broke with some of the renaming

### CODE REVIEW

**Review by Jesster2829 — APPROVED** [review]
hmmm not enough docstring I think...

Looks good

File: my-webxr-app/src/components/Button.tsx [hunk 1]
```diff
@@ -3,9 +3,14 @@ import { Interactive } from '@react-three/xr';
...
 import Box from './Box';
 
-// the useState hook is used to change the color of the box
-// the useState hook is used to change the scale of the box
-// taken from the Webxr example
+/**
+ * Returns an interactive button for VR space
+ * @pre-condition None
+ * @post-condition Returns an interactive button for VR space
+ * @param {JSX.IntrinsicElements["mesh"]} props
```
Comment (Subzeero) [inline my-webxr-app/src/components/Button.tsx:10]:
```suggestion
 * @param {JSX.IntrinsicElements["mesh"] | undefined} props
```

File: my-webxr-app/src/components/Box.tsx [hunk 2]
```diff
@@ -11,6 +11,20 @@ type BoxProps = {
...
+ * Returns a Box react element
+ * @pre-condition None
+ * @post-condition A box geometry for use in VR
+ * @param {Color | number} color
+ * @param {[width: number | undefined, height: number | undefined, depth: number | undefined,
+ *    widthSegments: number | undefined, heightSegments: number | undefined,
+ *    depthSegments: number | undefined] | undefined} size the size props for the box
+ * @param {[number, number, number]} scale the world scale of the box
+ * @param {JSX.Element | undefined} children child elements of the box
+ * @param {Omit<BoxProps, "scale" | "color" | "size" | "children">} rest other parameters
```
Comment (Subzeero) [inline my-webxr-app/src/components/Box.tsx:24]: This parameter is optional, so should we mark it as potentially undefined?
```suggestion
 * @param {Omit<BoxProps, "scale" | "color" | "size" | "children"> | undefined} rest other parameters
```

File: my-webxr-app/src/repository/Column.tsx [hunk 3]
```diff
@@ -8,6 +8,13 @@ export default class Column<Type> {
 
   values: Type;
 
+  /**
+   * Stores a CSV column
+   * @pre-condition Name is a string and values are strings or numbers
+   * @post-condition The column with the provided name will be represented by this object
+   * @param {string} name a string the is hte name of the column
```
Comment (Subzeero) [inline my-webxr-app/src/repository/Column.tsx:15]:
```suggestion
   * @param {string} name a string the is the name of the column
```

**Review by Subzeero — APPROVED** [review]
Gruelling work 🫡 looks good

**Pull request merged**